source/Halibut/Transport/SecureConnection.cs

```diff
@@ -36,6 +37,7 @@ public bool HasExpired()
 
         public void Dispose()
         {
+            OnDisposed?.Invoke(this, EventArgs.Empty);
```

Comment:
So should `OnDisposed?.Invoke` be called at the end of this method, given is past tense?